src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
```

Comment:
The use of `Objects` utility class is against the task checklist requirements. Please implement the equality check manually instead of using `Objects.equals`.